puppet/modules/eayunstack/manifests/upgrade/nova.pp

```diff
@@ -50,8 +52,35 @@
       onlyif => "match cinder/admin_username[.=\"${admin_username}\"] size < 1",
     }
 
-    Augeas['add_cinder_admin_info'] ~> Service['openstack-nova-api']
-    Augeas['add_cinder_admin_info'] ~> Service['openstack-nova-conductor']
+    augeas { 'set_reclaim_instance_interval':
+      context => '/files/etc/nova/nova.conf',
+      lens    => 'Puppet.lns',
+      incl    => '/etc/nova/nova.conf',
+      changes => [
+        "set DEFAULT/reclaim_instance_interval ${reclaim_instance_interval}",
+      ],
+      onlyif => "match DEFAULT/reclaim_instance_interval[.=\"${reclaim_instance_interval}\"] size < 1",
```

Comment:
哦，你设置的几个 neutron 参数都是可以指定多次的，我没注意到这点。
可是${reclaim_instance_interval} 是由我们指定的一个确定值，如果配置文件里有
reclaim_instance_interval=xxx，只要 xxx 跟我们指定的不一致就会被设置，一致则不会设置，似乎没有问题吧。会出现
xxx跟我们需要的值不一致，但 augeas 却跳过的情况吗？

On Tue, Mar 15, 2016 at 4:10 PM, huntxu <notifications@github.com> wrote:

> In puppet/modules/eayunstack/manifests/upgrade/nova.pp
> <https://github.com/eayunstack/eayunstack-upgrade/pull/19#discussion_r56126913>
> :
>
> > @@ -50,8 +52,35 @@
> >        onlyif => "match cinder/admin_username[.=\"${admin_username}\"] size < 1",
> >      }
> >
> > -    Augeas['add_cinder_admin_info'] ~> Service['openstack-nova-api']
> > -    Augeas['add_cinder_admin_info'] ~> Service['openstack-nova-conductor']
> > +    augeas { 'set_reclaim_instance_interval':
> > +      context => '/files/etc/nova/nova.conf',
> > +      lens    => 'Puppet.lns',
> > +      incl    => '/etc/nova/nova.conf',
> > +      changes => [
> > +        "set DEFAULT/reclaim_instance_interval ${reclaim_instance_interval}",
> > +      ],
> > +      onlyif => "match DEFAULT/reclaim_instance_interval[.=\"${reclaim_instance_interval}\"] size < 1",
>
> 需要确认一下这个地方的nova.conf是不是默认配置文件里面总是不会设置，如果是的话倒是可以先合并。
>
> 我倒是建议(以后再提交个patch)把这个onlyif去掉，因为这里这个reclaim_instance_interval不是一个可以指定多次的参数，当原本的配置文件里有reclaim_instance_interval=XXXX的时候这个augeas就直接跳过了可能不能得到正确结果。
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/eayunstack/eayunstack-upgrade/pull/19/files#r56126913
>



-- 
Regards,
apporc